alibi_detect/cd/utils.py

```diff
@@ -39,3 +43,67 @@ def update_reference(X_ref: np.ndarray,
             raise KeyError('Only `reservoir_sampling` and `last` are valid update options for X_ref.')
     else:
         return X_ref
+
+
+def activate_train_mode_for_dropout_layers(model: Union[nn.Module, tf.keras.Model], backend: str) -> Callable:
```

Comment:
There could be a few issues here:
- Since PyTorch is not a dependency, it might crash at import. Check https://github.com/SeldonIO/alibi-detect/blob/master/alibi_detect/cd/base.py#L12 and https://github.com/SeldonIO/alibi-detect/blob/master/alibi_detect/cd/base.py#L339 how this is handled in the base class.
- More importantly, this function contains backend specific functionality but is located in `alibi_detect.cd` and not e.g. `alibi_detect.cd.pytorch` or `alibi_detect.cd.tensorflow`. We should try to keep anything backend-specific in the relevant directories. As a result, the `activate_train_mode_for_dropout_layers` could be removed here and the backend-specific functionality could be invoked at initialization of the detector in `alibi_detect.cd.model_uncertainty.py` because it only needs to be done once. The backend-specific dropout activation could then simply live in `alibi_detect.utils.pytorch.utils.py` and `alibi_detect.utils.tensorflow.utils.py` since the functionality is not drift detector specific.